src/Toml/Parser.hs

```diff
@@ -79,12 +80,66 @@ basicStringP :: Parser Text
 basicStringP = lexeme $ Text.pack <$> (char '"' *> anyChar `manyTill` char '"')
 
 textP :: Parser Text
-textP = literalStringP <|> basicStringP
+textP = multilineStringP <|> multilineLiteralStringP <|> literalStringP <|> basicStringP
 
 -- adds " or ' to both sides
 quote :: Text -> Text -> Text
 quote q t = q <> t <> q
 
+nonControlChar :: Parser Text
+nonControlChar = Text.singleton <$> satisfy (not . isControl)
+
+multilineStringP :: Parser Text
+multilineStringP = lexeme $ fmap mconcat $ quotes3
+                                        >> optional eol
+                                        >> allowedChar `manyTill` quotes3
+  where
+    quotes3 = string "\"\"\""
+
+    allowedChar :: Parser Text
+    allowedChar = lineEndingBackslash <|> escapeSequence <|> nonControlChar <|> eol
+
+    lineEndingBackslash :: Parser Text
+    lineEndingBackslash = Text.empty <$ try (char '\\' >> eol >> space)
+
+    escapeSequence :: Parser Text
+    escapeSequence = char '\\' >> anyChar
+                 >>= \case
+                      'b'  -> return "\b"
```

Comment:
Oh, I see that you're already done issue #40. And this is very cool! So it's better to refactor code to make `basicStringP` parser also support these characters. But let's do this refactoring under separate PR, let's keep this PR simpler.

Reply:
That's why I wanted to merge #71 and close #40 before working on multiline strings.

But since we're working on #15 in this PR, what do you suggest as a change here? Should I make the implementation of `multilineStringP` wrong? or should I fix the implementation of `basicStringP`?

Reply:
Let's keep `basicStringP` and `multilineBasicStringP` as it is for now, and you will refactor this under your PR #71. So #71 will be a very easy PR 🙂